### DISCUSSION TITLE
Just created a runtime to make django gis/geo work in vercel

### DISCUSSION THREAD

**Discussion by jperelli** [vercel/vercel#8234] [Show and tell]
It took a long week of trying to build and rebuild the gdal/geos libraries and dependencies and make it work in vercel, but finally it's working.

You can use a runtime in vercel now called vercel-python-gis@1.0.0

You can see more info (like how to configure the rest of django to work in vercel) in the project page here https://github.com/jperelli/vercel-python-gis

Happy coding!

**Comment by leerob** [comment]
This is really neat 😄 Would love to hear more about what you're building with this.

Maybe if you're interested, we could make it a community runtime? https://github.com/vercel-community

**Reply by jperelli** [reply]
Hi @leerob!, thanks!

I'm building a directory of beaches.
I use GeoDjango to store beach geographic areas and administrative areas, and make some operations on those.
The website shows a map with the beach, flickr photos, climate, webcams and reviews using geolocation.
It's not ready yet, when I'm ready I'll publish it.

How would it be to make it a community runtime?

vercel-python-gis is a carbon copy of @python/vercel with an additional offline build script of geos/gdal/proj libraries that are stripped down to fit in a lambda function.
I'm thinking if there is a better way to do it like inherit from @python/vercel somehow so there's no duplicated code. And also automating build with gh actions maybe?

**Comment by michaelhenry** [comment]
@jperelli I'm new of using vercel. Just wondering how to use this inside my vercel.json. thanks

**Reply by jperelli** [reply]
Hi @michaelhenry you can see how to use it in the readme here https://github.com/jperelli/vercel-python-gis
There's a full example with Django.

**Reply by michaelhenry** [reply]
Thanks @jperelli . I was able to run it, but unfortunately it reached the max app size which is 250MB, do you know if vercel has some concept similar to AWS lambda layers? Thanks again.

**Reply by jperelli** [reply]
The whole idea of the runtime was to avoid this limit. When building the .o libraries, it strips all the debug info and other things to make them smaller to fit the lambda limit of 250MB. Maybe some dependency broke. I'll have to check it again.

**Comment by smazharzaidi** [comment]
```
(myenv) PS D:\University\FYP\final_project\land_management_system> vercel --prod
Vercel CLI 34.1.9
🔍  Inspect: https://vercel.com/mazhars-projects-cd008bc7/land-management-system/VmvTazbZAoA7vT19LpJznnPhJRjE [4s]
✅  Production: https://land-management-system-nzw9capu3-mazhars-projects-cd008bc7.vercel.app [4s]        
Error: Command failed with ENOENT: pip3.9 install --disable-pip-version-check --target . werkzeug==1.0.1 
spawn pip3.9 ENOENT
```
I'm getting this error, when I run this command in terminal, it's running, but it's not running with vercel --prod
